src/app/common/DataTabel/Tabel.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import React from "react";
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+import { Member } from "@/types";
+import useGroupedData from "@/hooks/use-group-data";
+
+type DataTableProps = {
+  data: Member[];
+  columns: any;
+  isGroupData: boolean;
+};
+
+export const DataTabel = ({ data, columns, isGroupData } : DataTableProps) => {
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  const groupedData = useGroupedData(data, 'shift');
+
+  return (
+    <div className="container mx-auto mt-4">
+      <Table>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <TableHead key={header.id}>
+                  {flexRender(header.column.columnDef.header, header.getContext())}
+                </TableHead>
+              ))}
+            </TableRow>
+          ))}
+        </TableHeader>
+
+        <TableBody>
+          {isGroupData
+            ? Object.entries(groupedData).map(([shift, rows]) => (
```

Comment:
i think we can better data structure to avoid map on object.entries

Reply:
I've adjusted the structure so that the output is now in the form of an array of objects.

Reply:
I have adjusted the structure so that the output is now in the form of an array of objects.

Reply:
We should also rename that, `groupData` isn't a good name and I think we'd need a memoization here as well.

Reply:
You can get rid of shift column too because it's already a group header.